packages/react-dom-bindings/src/client/ReactDOMComponent.js

```diff
@@ -2485,51 +2485,41 @@ function diffHydratedGenericElement(
       case 'src':
       case 'href':
         if (
+          __DEV__ &&
```

Comment:
The entire `diffHydratedGenericElement` function is only called in dev so this is really unnecessary.

Reply:
I think we only have it due to the lint rule about console.error doesn't know. You could probably also early return in the beginning of this file if it's not dev

Reply:
I see now, this is just to silence the console.error linter